### PR TITLE
Replace is integer

### PR DESCRIPTION
**Pull request by hjmjohnson**
Prefer built-in C++11 language features for identifying signed types.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) was made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

File: Modules/Core/Common/include/itkMath.h [hunk 1]
```diff
@@ -516,9 +516,9 @@ template <typename TInputType1, typename TInputType2>
 struct AlmostEqualsScalarImplementer
 {
   static constexpr bool TInputType1IsInteger = std::is_integral<TInputType1>::value;
-  static constexpr bool TInputType1IsSigned = itk::NumericTraits<TInputType1>::IsSigned;
+  static constexpr bool TInputType1IsSigned = std::is_signed<TInputType1>::value;
```
Comment (dzenanz) [inline Modules/Core/Common/include/itkMath.h:519]: In commit message, `IsInteger` should be `IsSigned`. e.g. `STYLE: Remove IsInteger and replace with std::is_signed` -> `STYLE: Remove NumericTraits::IsSigned and replace with std::is_signed`.

File: Modules/Core/Common/test/itkNumericTraitsTest.cxx [hunk 2]
```diff
@@ -205,22 +203,22 @@ CheckSignedAndIntegerTraitsSameAsSTDNumericLimits(const char * const name)
   bool didTestPass = true;
   std::cout << "    " << name << std::endl;
   // test for IsSigned
-  if (itk::NumericTraits<T>::IsSigned != std::numeric_limits<T>::is_signed)
```
Comment (dzenanz) [inline Modules/Core/Common/test/itkNumericTraitsTest.cxx:208]: This line should not be modified.

File: Modules/Core/Common/test/itkNumericTraitsTest.cxx [hunk 3]
```diff
@@ -268,14 +266,14 @@ CheckSignedAndIntegerTraitsForComplexTypes(const char * const name)
   }
 
   // IsSigned same for complex and basic types??
-  if (itk::NumericTraits<T>::IsSigned != itk::NumericTraits<typename itk::NumericTraits<T>::ValueType>::IsSigned)
```
Comment (dzenanz) [inline Modules/Core/Common/test/itkNumericTraitsTest.cxx:268]: This line should not be changed.

File: Modules/Core/Common/test/itkNumericTraitsTest.cxx [hunk 4]
```diff
@@ -360,7 +358,7 @@ CheckIsComplexTraits()
   std::cout << "Testing non complex types for IsComplex trait" << std::endl;
   if (itk::NumericTraits<float>::IsComplex || itk::NumericTraits<double>::IsComplex ||
       itk::NumericTraits<char>::IsComplex || itk::NumericTraits<int>::IsComplex ||
-      itk::NumericTraits<unsigned long>::IsComplex)
+      itk::NumericTraits<unsigned long>::IsComplex || itk::NumericTraits<float>::IsInteger)
```
Comment (dzenanz) [inline Modules/Core/Common/test/itkNumericTraitsTest.cxx:361]: All the checks here are for `IsComplex`. This line should not be changed.

**Review by dzenanz — COMMENTED** [review]
Also, first commit in this PR fails to compile, as pointed out in its PR.

File: Modules/Core/Common/include/itkConceptChecking.h [hunk 5]
```diff
@@ -852,7 +852,7 @@ template <typename T>
 struct IsInteger
 {
   using Self = IsInteger;
-  static constexpr bool Integral = NumericTraits<T>::IsInteger;
+  static constexpr bool Integral = std::is_integral<T>::value;
```
Comment (dzenanz) [inline Modules/Core/Common/include/itkConceptChecking.h:855]: Commit message's title should be changed to `STYLE: Remove uses of IsInteger and replace with std::is_integral`

Comment (hjmjohnson) [reply Modules/Core/Common/include/itkConceptChecking.h:855]: Fixed.

File: Modules/Core/Common/test/itkNumericTraitsTest.cxx (hunk 3 above)
Comment (hjmjohnson) [reply Modules/Core/Common/test/itkNumericTraitsTest.cxx:268]: This must be changed or lines 68 and 69 need to be resored.

File: Modules/Core/Common/include/itkConceptChecking.h (hunk 5 above)
Comment (N-Dekker) [inline Modules/Core/Common/include/itkConceptChecking.h:855]: C++14 offers a few alternatives to the form `std::is_integral<T>::value`, like:

    static constexpr bool Integral2 = std::is_integral<T>();
    static constexpr bool Integral3 = std::is_integral<T>{};
    static constexpr bool Integral4 = std::is_integral<T>{}();

But I think `std::is_integral<T>::value` is just fine, here 👍

**Review by dzenanz — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Pull request merged**